### ISSUE TITLE
Static storage variables resetting after reloading dll module

### ISSUE THREAD

**Issue by acarx** [ocornut/imgui#1105]
I'm using a dll to update code without restarting my application. "imgui.h" mentiones this but doesn't go into detail. I was expecting there to be some static storage variables but it seems that there are more statically storred variables within those. And since these don't have a proper copy constructor I can't just save them before unloading and load after.

So my question would be:
What exactly do I need to create in order to safely unload a module and restore after reloading that module?

**Comment by AcarX** [comment]
Temporary solution I came up with was to setup ImGui every time I reload the dll. It's likely to be causing some small memory leaks but it seems to be working overall.

**Comment by ocornut** [comment]
Hello AcarX, sorry for my late answer.

Calling Shutdown() before reloading should re-init and the window position/size should be preserved (unless flagged with `ImGuiWindowFlags_NoSavedSettings`. The problem is that you will lose the window storage namely the open/closed state of tree nodes and collapsing headers.

What you can do is create your own ImGuiContext + ImFontAtlas explicitly, assign the font atlas to the context, and carry on those without using the default static contexts. That should solve your problems.

**Comment by ocornut** [comment]
In the absence of an answer from @acarx , closing this as solved.
Adding link to #586 for reference.
TL;DR; 
Create your own context `CreateContext`+`SetCurrentContext`, and font atlas `context->GetIO().Fonts = new FontAtlas()`.
